crates/ide-assists/src/handlers/change_visibility.rs

```diff
@@ -46,16 +47,36 @@ fn add_vis(acc: &mut Assists, ctx: &AssistContext<'_>) -> Option<()> {
 
     let (offset, target) = if let Some(keyword) = item_keyword {
         let parent = keyword.parent()?;
-        let def_kws =
-            vec![CONST, STATIC, TYPE_ALIAS, FN, MODULE, STRUCT, ENUM, TRAIT, USE, MACRO_DEF];
-        // Parent is not a definition, can't add visibility
-        if !def_kws.iter().any(|&def_kw| def_kw == parent.kind()) {
+
+        let can_add = |node: &SyntaxNode| -> bool {
```

Comment:
`can_add` doesn't need to be a closure I'd say. Can you make this a proper function or inline the body even as it's only used once?